src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -321,6 +321,16 @@ public override Task<bool> HandleRequestAsync()
                 _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
             }
 
+            if (properties.Items.TryGetValue(OpenIdConnectDefaults.PromptPropertiesKey, out var prompt))
+            {
+                // remove consumed item from properties to avoid serializing it as state
+                properties.Items.Remove(OpenIdConnectDefaults.PromptPropertiesKey);
```

Comment:
This has a bit of a dirty feel to it, do we stash/remove things from the authentication properties anywhere else?

Reply:
I took that idea from the [GoogleHandler](https://github.com/aspnet/Security/blob/rel/2.0.0/src/Microsoft.AspNetCore.Authentication.Google/GoogleHandler.cs#L87). Not sure if that’s really necessary or not. But it does make a bit sense to me to clean out those properties from the serialized state which will end up in the URL anyway.